spec/rest-middleware.spec.js

```diff
@@ -0,0 +1,84 @@
+"use strict";
+
+describe("restMiddleware", () => {
+    var configMock, helmetMock, logger, restMiddleware, serverMock, restifyLinks;
+
+    beforeEach(() => {
+        var LoggerMock, RestMiddleware;
+
+        RestMiddleware = require("../lib/rest-middleware");
+        LoggerMock = require("./mock/logger-mock");
+        helmetMock = jasmine.createSpyObj("helmetMock", [
+            "frameguard",
+            "hidePoweredBy",
+            "hsts",
+            "ieNoOpen",
+            "noCache",
+            "noSniff",
+            "xssFilter"
+        ]);
+        serverMock = jasmine.createSpyObj("serverMock", [
+            "use"
+        ]);
+        configMock = {
+            baseUrl: "http://localhost:8443",
+            https: false
+        };
+        restifyLinks = jasmine.createSpy();
+        logger = new LoggerMock();
+        restMiddleware = new RestMiddleware(helmetMock, logger, restifyLinks);
+    });
+
+    function expectNormalMiddlewareWasCalled () {
+        expect(helmetMock.frameguard).toHaveBeenCalled();
+        expect(helmetMock.ieNoOpen).toHaveBeenCalled();
+        expect(helmetMock.hidePoweredBy).toHaveBeenCalled();
+        expect(helmetMock.ieNoOpen).toHaveBeenCalled();
+        expect(helmetMock.noCache).toHaveBeenCalled();
+        expect(helmetMock.noSniff).toHaveBeenCalled();
+        expect(helmetMock.xssFilter).toHaveBeenCalled();
+        expect(restifyLinks).toHaveBeenCalled();
+    }
+
+    it("calls restMiddleware without https", () => {
+        restMiddleware(configMock, serverMock);
+
+        expectNormalMiddlewareWasCalled();
+        expect(helmetMock.hsts).not.toHaveBeenCalled();
+    });
+    it("calls restMiddleware with https", () => {
+        configMock.https = true;
+        restMiddleware(configMock, serverMock);
+
+        expectNormalMiddlewareWasCalled();
+        expect(helmetMock.hsts).toHaveBeenCalled();
+    });
+    it("sets up link discovery", () => {
+        var next, req, res, standardLinks;
+        res = jasmine.createSpyObj("resMock", [
+            "links"
+        ]);
+        req = {
+            href: function () {
+                return "/path"
+            }
+        }
+        spyOn(req, "href").andCallThrough();
```

Comment:
We want to leverage as much Jasmine as we can.

    req = jasmine.createSpyObj("req", [
        "href"
    ]);
    req.href.andReturn("/path");